app/models/concerns/sluggable.rb

```diff
@@ -37,28 +37,21 @@ def slug
     @slug ||= persisted_slug
   end
 
+  def computed_slug 
+    Slugs::ComputeSlug.call(slug_prefix: self.class.slug_prefix, slug: slug)
+  end
+
   private
 
   def persisted_slug
     @persisted_slug = slugs.find_by(active: true).try(:computed_slug)
   end
 
   def update_slug_history
-    slug_created = self.class.transaction do
-      self.slugs.where(active: true).update_all(active: false)
-      self.slugs.build(slug_prefix: self.class.slug_prefix, 
-                       slug: slug, 
-                       computed_slug: compute_slug)
-    end
-
-    unless slug_created
-      self.errors.add(:slug, "could not be saved, already taken")
+    unless Slugs::UpdateSluggableHistory.call(slug_prefix: self.class.slug_prefix, resource: self)
+      self.errors.add(:slug, 'already taken')
+      throw :abort # Rails 5 only
+      raise ActiveRecord::Rollback
```

Comment:
Unless you've left it as the throw will break early in Rails 5, then when we merge to Shift, we remove the throw?